test/eventHandlers/invocationRequest.test.ts

```diff
@@ -169,7 +169,7 @@ namespace Msg {
             invocationId: '1',
             message: "Error: Choose either to return a promise or call 'done'.  Do not use both in your script.",
             level: LogLevel.Error,
-            logCategory: LogCategory.User,
+            logCategory: LogCategory.System,

```

Comment:
Why are we changing the log category? I assume the errors will go to different places.

Reply:
This was not necessarily a planned change, but I would say my refactor uncovered a "bug" and my easiest course of action was to fix the bug by changing the category here.

To me, the distinction is:
- If the message came from the user, it's a "User" log
- If the message came from us, it's a "System" log

One key difference is I believe we don't track user logs in our own telemetry, just system logs. I think having these particular errors in our own telemetry would be very useful for tracking purposes, detectors, etc.

Reply:
yes, it's useful for us as well as for the user - user also need a way to be notified about the issue. Can we add the error to both - user and system?

Reply:
I think that would result in duplicate logs. In all my testing, system logs are also shown to the user. In fact, the vast majority of the logs in this repo (15 out of 17ish) were already categorized as `System` (for example, "incorrect Node.js version" [here](https://github.com/Azure/azure-functions-nodejs-worker/blob/v3.x/src/eventHandlers/workerInitRequest.ts#L27) and "Failed to load function" [here](https://github.com/Azure/azure-functions-nodejs-worker/blob/v3.x/src/eventHandlers/functionLoadRequest.ts#L24))

I honestly think the only reason we have the distinction is so that we don't track user logs in our telemetry to protect their privacy. I don't think it affects what gets shown

Reply:
> for example, "incorrect Node.js version" [here](https://github.com/Azure/azure-functions-nodejs-worker/blob/v3.x/src/eventHandlers/workerInitRequest.ts#L27) and "Failed to load function" [here](https://github.com/Azure/azure-functions-nodejs-worker/blob/v3.x/src/eventHandlers/functionLoadRequest.ts#L24)

the logs are [console logs](https://github.com/Azure/azure-functions-nodejs-worker/blob/v3.x/src/utils/Logger.ts#L5) which "guarantee the messages reach the host despite potential performance impact."

> I think that would result in duplicate logs. In all my testing, system logs are also shown to the user.
I am wandering what is the difference between grpc channel logs "User" and "System" in app insights? Are the logs shown differently to the customer?

Reply:
> the logs are [console logs](https://github.com/Azure/azure-functions-nodejs-worker/blob/v3.x/src/utils/Logger.ts#L5) which "guarantee the messages reach the host despite potential performance impact."

Sorry, the first link is a console log, but the second link for "function load failure" is a channel.log.

I'm not seeing much difference at all in app insights. For example here's some info I see in the portal in the app insights "Live metrics" blade:
|user|system|
|---|---|
|![Screen Shot 2022-03-08 at 6 38 14 PM](https://user-images.githubusercontent.com/11282622/157362026-f02f8b5e-3498-4c82-818a-90b49140c4ae.png)|![Screen Shot 2022-03-08 at 6 38 27 PM](https://user-images.githubusercontent.com/11282622/157362029-265d63b1-8a3b-4bfe-8e33-c1f46f0cd520.png)|

The only other difference I can think of is how logs get filtered. [This repo](https://github.com/anthonychu/functions-log-suppression) has a good description on that. I believe "User" logs would translate to the "Function.<function_name>.User" category and our system logs would translate to the "Worker.rpcWorkerProcess" category. All of these can be individually filtered by the user so there's a chance they wouldn't see the log if they explicitly turned off all Worker logs, but by default all worker errors and warnings (aka the logs being discussed in this thread) are shown

Reply:
As discussed offline, we discovered some of these logs are currently broken _before_ my changes. I undid my behavior change to the LogCategory and we will fix the problem separately. Issue in the host is here: https://github.com/Azure/azure-functions-host/issues/8222